agent/lib_php_amqplib.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * Functions relating to instrumenting the AWS-SDK-PHP.
+ * https://github.com/aws/aws-sdk-php
+ */
+#include "php_agent.h"
+#include "php_call.h"
+#include "php_hash.h"
+#include "php_wrapper.h"
+#include "fw_hooks.h"
+#include "fw_support.h"
+#include "util_logging.h"
+#include "lib_php_amqplib.h"
+
+#define PHP_PACKAGE_NAME "php-amqplib/php-amqplib"
+
+/*
+ * nr_php_amqplib_handle_version will automatically load the class if it isn't
+ * loaded yet and then evaluate the string. To avoid the VERY unlikely but not
+ * impossible fatal error if the file/class isn't loaded yet, we need to wrap
+ * the call in a try/catch block and make it a lambda so that we avoid fatal
+ * errors.
+ */
+void nr_php_amqplib_handle_version() {
+  char* version = NULL;
+  zval retval;
+  int result = FAILURE;
+
+  result = zend_eval_string(
+      "(function() {"
+      "     $nr_php_amqplib_version = '';"
+      "     try {"
+      "          $nr_php_amqplib_version = PhpAmqpLib\\Package::VERSION;"
+      "     } catch (Throwable $e) {"
+      "     }"
+      "     return $nr_php_amqplib_version;"
+      "})();",
+      &retval, "Get nr_php_amqplib_version");
+
+  /* See if we got a non-empty/non-null string for version. */
+  if (SUCCESS == result) {
```

Comment:
What is the condition to detect when the `catch` was triggered in the lambda function?  What will be the value of `result` and of `retval` in that situation?

Reply:
result will be success.

Reply:
Could we change it so we return empty string explicitly so it is not relying on a current behavior of PHP and so it is more clear how this code interacts with the calling C code?

Reply:
I understand the code now - the initialized value (empty string) is used as the `catch` doesn't actually need to modify its value.